cardano-api/src/Cardano/Api/Crypto/Bip32Ed25519.hs

```diff
@@ -0,0 +1,204 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- | BIP32-Ed25519 digital signatures.
+module Cardano.Api.Crypto.Bip32Ed25519
+  ( Bip32Ed25519DSIGN
+  , SigDSIGN (..)
+  , SignKeyDSIGN (..)
+  , VerKeyDSIGN (..)
+
+    -- * Serialisation
+  , xPrvToBytes
+  , xPrvFromBytes
+  )
+where
+
+import           Cardano.Prelude hiding (show)
+import           Prelude (show)
+
+import           Data.ByteArray as BA (ByteArrayAccess, ScrubbedBytes, convert)
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+import           GHC.Generics (Generic)
+
+import           Cardano.Binary (FromCBOR (..), ToCBOR (..))
+import qualified Cardano.Crypto.Wallet as CC
+
+import           Cardano.Crypto.DSIGN.Class
+import           Cardano.Crypto.Seed
+import           Cardano.Crypto.Util (SignableRepresentation (..))
+
+import qualified Crypto.ECC.Edwards25519 as Ed25519
+import           Crypto.Error (eitherCryptoError)
+
+
+data Bip32Ed25519DSIGN
+
+instance DSIGNAlgorithm Bip32Ed25519DSIGN where
+
+    --
+    -- Key and signature types
+    --
+
+    newtype VerKeyDSIGN Bip32Ed25519DSIGN = VerKeyBip32Ed25519DSIGN CC.XPub
+        deriving (Show, Eq, Generic)
+        deriving newtype NFData
+        deriving NoUnexpectedThunks via UseIsNormalForm CC.XPub
+
+    newtype SignKeyDSIGN Bip32Ed25519DSIGN = SignKeyBip32Ed25519DSIGN CC.XPrv
+        deriving (Generic, ByteArrayAccess)
+        deriving newtype NFData
+        deriving NoUnexpectedThunks via UseIsNormalForm CC.XPrv
+
+    newtype SigDSIGN Bip32Ed25519DSIGN = SigBip32Ed25519DSIGN CC.XSignature
+        deriving (Show, Eq, Generic, ByteArrayAccess)
+        deriving NoUnexpectedThunks via UseIsNormalForm CC.XSignature
+
+    --
+    -- Metadata and basic key operations
+    --
+
+    algorithmNameDSIGN _ = "bip32_ed25519"
+
+    deriveVerKeyDSIGN (SignKeyBip32Ed25519DSIGN sk) =
+      VerKeyBip32Ed25519DSIGN $ CC.toXPub sk
+
+    --
+    -- Core algorithm operations
+    --
+
+    type Signable Bip32Ed25519DSIGN = SignableRepresentation
+
+    signDSIGN () a (SignKeyBip32Ed25519DSIGN sk) =
+      SigBip32Ed25519DSIGN $
+        CC.sign (mempty :: ScrubbedBytes) sk (getSignableRepresentation a)
+
+    verifyDSIGN () (VerKeyBip32Ed25519DSIGN vk) a (SigBip32Ed25519DSIGN sig) =
+      if CC.verify vk (getSignableRepresentation a) sig
+        then Right ()
+        else Left "Verification failed"
+
+    --
+    -- Key generation
+    --
+
+    seedSizeDSIGN _  = 32
+
+    genKeyDSIGN seed =
+      SignKeyBip32Ed25519DSIGN $
+        CC.generateNew
+          (getSeedBytes seed)
+          (mempty :: ScrubbedBytes)
+          (mempty :: ScrubbedBytes)
```

Comment:
The current code in the cardano-api uses `Crypto.HD.generate` rather than `Crypto.HD.generateNew`. We should probably not change that without reason. We can check with the wallet team which one of the two they use.

Reply:
Ok so the wallet does use `Crypto.HD.generateNew`. :confused:

Reply:
`generateNew` is the "new" way of generating root/master keys.

Not only does `cardano-wallet` use it, but `cardano-addresses` also uses this method of generation for both Icarus and Shelley-style extended keys.